### PR TITLE
Add team, add folder, delete team, leave team

### PR DESCRIPTION
**Pull request by srkirkland**
News pages for adding teams and folders.  Delete team is a modal that sets team to inactive.  Leave team removes your permissions but if you are the only admin for a team it just sets that whole team to inactive too.

### CODE REVIEW

**Review by jSylvestre — APPROVED** [review]

File: Finjector.Web/ClientApp/src/components/Teams/DeleteTeam.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      onSuccess: () => {
+        navigate("/teams");
+        toggleModal();
+      },
+    });
+  };
+
+  return (
+    <>
+      <button className="btn btn-new me-3" onClick={toggleModal}>
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/DeleteTeam.tsx:32]: should be `<Button>`, you use it in the footer

File: Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      onSuccess: () => {
+        navigate("/teams");
+        toggleModal();
+      },
+    });
+  };
+
+  return (
+    <>
+      <button className="btn btn-new me-3" onClick={toggleModal}>
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx:33]: also `<Button>`

File: Finjector.Web/ClientApp/src/components/Teams/NameAndDescriptionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          type="text"
+          className="form-control"
+          id="name"
+          name="name"
+          value={formData.name}
+          onChange={handleChange}
+          required
+          maxLength={50}
+        />
+        <div className="invalid-feedback">Please provide a name.</div>
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/NameAndDescriptionForm.tsx:41]: i don't ever see this. just get this: 
image
also, should these all be Reactstrap?

File: Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const leaveMutation = useLeaveTeamMutation();
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+
+  const handleDelete = () => {
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx:22]: handleLeave

File: Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            </p>
+          )}
+        </ModalBody>
+        <ModalFooter>
+          <Button
+            color="danger"
+            onClick={handleDelete}
+            disabled={leaveMutation.isLoading}
+          >
+            Delete
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/LeaveTeam.tsx:53]: should be Leave

File: Finjector.Web/ClientApp/src/components/Teams/NameAndDescriptionForm.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          id="description"
+          name="description"
+          value={formData.description}
+          onChange={handleChange}
+          maxLength={300}
+        />
+        <div className="invalid-feedback">Please provide a description.</div>
+      </div>
+
+      <button type="submit" className="btn btn-primary" disabled={loading}>
```
Comment (laholstege) [inline Finjector.Web/ClientApp/src/components/Teams/NameAndDescriptionForm.tsx:59]: Button?

**Review by laholstege — COMMENTED** [review]
pattern is good. some inconsistencies in naming & such, we can blame copilot for that :^)

**Pull request merged**